### PR TITLE
Clipping deformer & cosmetic fixes

### PR DESCRIPTION
**Pull request by jatinkhilnani**
- Clipping deformer (#1) 
- Minor corrections to documentation

### CODE REVIEW

**Comment by bmcfee** [comment]
It looks like this PR only includes the cosmetic fixes?

**Comment by jatinkhilnani** [comment]
- Included the deformer code `clipping.py` with tests
- Have implemented the test parameterization both in the current and new syntax, that eliminates the warnings with `pytest`  latest version (5.4.3)
- Added comments for `test_deformers`
- Open item: Per new syntax, added expectation (type of error) raised for `xfail` markers for `Clipping` method. The same is not working for `LinearClipping` & `RandomClipping`

File: muda/deformers/clipping.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    This contains the deformation function 
+    but does not manage state or parameters.
+    """
+
+    def __init__(self):
+        BaseTransformer.__init__(self)
+
+        # Build the annotation mappers
+#         self._register(".*", self.deform_times)
+#         self._register("tempo", self.deform_tempo)
```
Comment (bmcfee) [inline muda/deformers/clipping.py:25]: no annotation mappers needed for this one

File: tests/test_deformers.py [hunk 2]
```diff
@@ -734,3 +767,142 @@ def test_base_transformer():
...
+
+        # Verify that the state and history objects are intact
+        __test_deformer_history(D, jam_new.sandbox.muda.history[-1])
+
+        d_state = jam_new.sandbox.muda.history[-1]['state']
+        d_clip_limit = d_state['clip_limit']
+        assert lower <= d_clip_limit <= upper
+
+        # Verify clipping outcome
+        __test_audio(jam_orig, jam_new, d_clip_limit)
```
Comment (bmcfee) [inline tests/test_deformers.py:859]: can we rename this helper function?  Many things could be "test audio", but this is specifically testing for clipping.

File: muda/deformers/clipping.py (hunk 1 above)
Comment (jatinkhilnani) [reply muda/deformers/clipping.py:25]: Missed removing this commented out portion, would take care of it.

File: tests/test_deformers.py (hunk 2 above)
Comment (jatinkhilnani) [reply tests/test_deformers.py:859]: Is `__test_clipped_buffer` okay?

File: muda/deformers/clipping.py (hunk 1 above)
Comment (jatinkhilnani) [reply muda/deformers/clipping.py]: Removed.

File: tests/test_deformers.py (hunk 2 above)
Comment (jatinkhilnani) [reply tests/test_deformers.py]: Changed.

**Comment by jatinkhilnani** [comment]
Along with changes per review comments, included the ones mentioned in comment above (not for older tests). Modified config to include `pytest-cov==2.9.0`, since v2.10.0 required `pytest>=4.6` creating a conflict and causing build failure.

This does not completely resolve the CI check issues due to `llvmlite>=0.31.0dev0->numba>=0.43.0->librosa>=0.4`, as both the former packages are now compatible only with Python>=3.6. I see that `librosa` CI checks are now only performed for compatible Python versions. I suppose that needs to be modified here as well, or is there a workaround?

**Comment by bmcfee** [comment]
Thanks @jatinkhilnani !  I'll take the dependency issue in a separate thread, but yeah, we're generally overdue for some modernization.

**Comment by jatinkhilnani** [comment]
Thank you for the same. I have created another PR #82 for filter deformer, can include modernization (#81) changes in that one or contribute separately as well.

**Pull request merged**